src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperContext.cs

```diff
@@ -46,6 +68,14 @@ public class TagHelperContext
         }
 
         /// <summary>
+        /// The HTML element's tag name.
+        /// </summary>
+        /// <remarks>
+        /// A whitespace or <c>null</c> value results in no start or end tag being rendered.
```

Comment:
The only time `TagName` will be null will be if a user constructs `TagHelperContext` in its old format.